### PR TITLE
Define an enum for the RSA padding scheme

### PR DESCRIPTION
**Pull request by WesleyRosenblum**
Description of changes:

Define an enum for the RSA padding scheme #160 

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

# Check any applicable:
- [ ] Were any files moved? Moving files changes their URL, which breaks all hyperlinks to the files.

### CODE REVIEW

File: src/main/java/com/amazonaws/encryptionsdk/keyrings/RawRsaKeyringBuilder.java [hunk 1]
```diff
@@ -97,6 +97,31 @@ public RawRsaKeyringBuilder wrappingAlgorithm(String wrappingAlgorithm) {
      * @return The {@link Keyring} instance
      */
     public Keyring build() {
-        return new RawRsaKeyring(keyNamespace, keyName, publicKey, privateKey, wrappingAlgorithm);
+        return new RawRsaKeyring(keyNamespace, keyName, publicKey, privateKey, paddingScheme);
+    }
+
+    public enum RsaPaddingScheme {
```
Comment (SalusaSecondus) [inline src/main/java/com/amazonaws/encryptionsdk/keyrings/RawRsaKeyringBuilder.java]: NitPick: For ease of use, calculate the correct OAEPParameterSpec instances for each of these (they are immutable) and just pull them from the enum.

File: src/main/java/com/amazonaws/encryptionsdk/keyrings/RawRsaKeyringBuilder.java [hunk 2]
```diff
@@ -97,6 +97,31 @@ public RawRsaKeyringBuilder wrappingAlgorithm(String wrappingAlgorithm) {
...
+
+        private final String transformation;
+
+        RsaPaddingScheme(String transformation) {
+            this.transformation = transformation;
+        }
+
+        /**
+         * The Cipher transformation standard name as specified in
+         * https://docs.oracle.com/javase/8/docs/technotes/guides/security/StandardNames.html#Cipher
```
Comment (SalusaSecondus) [inline src/main/java/com/amazonaws/encryptionsdk/keyrings/RawRsaKeyringBuilder.java]: This is *almost* correct. Please clarify that in all cases the hash function used with MGF1 is the same as the hash function used directly with the message.

Comment (WesleyRosenblum) [reply src/main/java/com/amazonaws/encryptionsdk/keyrings/RawRsaKeyringBuilder.java]: Updated

File: src/main/java/com/amazonaws/encryptionsdk/keyrings/RawRsaKeyringBuilder.java (hunk 1 above)
Comment (WesleyRosenblum) [reply src/main/java/com/amazonaws/encryptionsdk/keyrings/RawRsaKeyringBuilder.java]: I'll hold off on this until Master Keys are deleted

**Pull request merged**